sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/db/BaseDbWriter.java

```diff
@@ -141,6 +147,29 @@ public ResultSet executeQueryWithResultSet(String sql) throws SQLException {
 
     }
 
+    public String addTimeZoneToColumnDefinition(String typeName, String timeZone) {
```

Comment:
not sure it is a good idea to use a regexp here as the target DateTime64 may already have a TZ !

example  DateTime64(6,'UTC') becomes
  DateTime64(6,'America/Chicago')

Reply:
I wonder if it works if you use a String instead, CH will do the conversion

Exception I got with the maximum boundaries
```
`db_to` DateTime64(6,\'America/Chicago\'),`_version` UInt64,`is_deleted` UInt8')
118741 2023-11-13 02:53:56.165 [pool-1-thread-9] ERROR com.altinity.clickhouse.sink.connector.db.DbWriter  - ******* ERROR inserting Batch *****************
java.lang.IllegalArgumentException: DateTime(9904571999) should be between -1420070400 and 9904550399 inclusive of both values
	at com.clickhouse.data.ClickHouseChecker.newException(ClickHouseChecker.java:17)
	at com.clickhouse.data.ClickHouseChecker.between(ClickHouseChecker.java:109)
	at com.clickhouse.data.format.BinaryDataProcessor$DateTime64SerDe.serialize(BinaryDataProcessor.java:292)
	at com.clickhouse.data.ClickHouseDataProcessor.write(ClickHouseDataProcessor.java:534)
	at com.clickhouse.jdbc.internal.InputBasedPreparedStatement.addBatch(InputBasedPreparedStatement.java:345)
	at com.altinity.clickhouse.sink.connector.db.DbWriter.addToPreparedStatementBatch(DbWriter.java:479)
	at com.altinity.clickhouse.sink.connector.executor.ClickHouseBatchRunnable.flushRecordsToClickHouse(ClickHouseBatchRunnable.java:199)
	at com.altinity.clickhouse.sink.connector.executor.ClickHouseBatchRunnable.processRecordsByTopic(ClickHouseBatchRunnable.java:169)
	at com.altinity.clickhouse.sink.connector.executor.ClickHouseBatchRunnable.run(ClickHouseBatchRunnable.java:101)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Reply:
This is quite logical. We can take into account source.timezone for `checkIfDateExceedsSupportedRange()` methods.

Reply:
yes that works. @subkanthi FTI

Reply:
> not sure it is a good idea to use a regexp here as the target DateTime64 may already have a TZ !
> 
> example DateTime64(6,'UTC') becomes DateTime64(6,'America/Chicago')

Could you explain, please? I don't see any problem here (except possible slowness)). I specifically override it within `input()` function to make Clickhouse use defined `source.timezone` to parse incoming data instead of column metadata.

Reply:
It is the wrong place to change the schema. The sink-connector should create the schema with the correct timezone (target timezone).

Reply:
in the sink-connector, there are 2 places we fix the schema 
- for the initial snapshot via custom tools like https://github.com/Altinity/clickhouse-sink-connector/pull/375/files
- via debezium, the sink-connector needs to add this `clickhouse.datetime.timezone` parameter. In your case, it would be the same as the source. default should be not specified and default to `source.datetime.timezone`

Reply:
in terms of timing, let us wait for @subkanthi to merge all pending PRs to develop, we will then address this one.

Reply:
@subkanthi  let us implement this too https://github.com/Altinity/clickhouse-sink-connector/issues/349